clippy_lints/src/indexing_slicing.rs

```diff
@@ -146,7 +157,7 @@ impl<'tcx> LateLintPass<'tcx> for IndexingSlicing {
                 // Catchall non-range index, i.e., [n] or [n << m]
                 if let ty::Array(..) = ty.kind() {
                     // Index is a const block.
-                    if let ExprKind::ConstBlock(..) = index.kind {
+                    if self.suppress_lint_in_const && let ExprKind::ConstBlock(..) = index.kind {
                         return;
                     }
```

Comment:
The lint emission in line 155 and 170 both use `span_lint_and_help` and suggest using `.get(n)` or `.get_mut(n)`. This suggestion is not applicable in constant context. Could you use [`span_lint_and_then()`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/diagnostics/fn.span_lint_and_then.html) and construct a message using the diagnostic builder? In constant context, it would be good to either not emit the suggestion or add another note saying: "the suggestion might not be applicable in constant blocks". From these options, I prefer the second one.

Reply:
Do you mean replace `span_lint_and_help` with with [`span_lint_and_then()`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/diagnostics/fn.span_lint_and_then.html) in line 155 and 170?

Reply:
Yes, then you can include optional information in the closure. Maybe try to search clippy for an example of [`span_lint_and_then`](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/diagnostics/fn.span_lint_and_then.html)